cpp/include/cudf/interop.hpp

```diff
@@ -162,6 +164,54 @@ std::shared_ptr<arrow::Scalar> to_arrow(cudf::scalar const& input,
                                         column_metadata const& metadata = {},
                                         rmm::cuda_stream_view stream = cudf::get_default_stream(),
                                         arrow::MemoryPool* ar_mr = arrow::default_memory_pool());
+
+/**
+ * @brief Create ArrowSchema from cudf table and metadata
+ *
+ * Populates and returns an ArrowSchema C struct using a table and metadata.
+ *
+ * @note For decimals, since the precision is not stored for them in libcudf,
+ * decimals will be converted to an Arrow decimal128 which has the widest precision that cudf
+ * decimal type supports. For example, `numeric::decimal32` will be converted to Arrow decimal128
+ * with the precision of 9 which is the maximum precision for 32-bit types. Similarly,
+ * `numeric::decimal128` will be converted to Arrow decimal128 with the precision of 38.
+ *
+ * @param input table_view to create a schema from
+ * @param metadata Contains the hierarchy of names of columns and children
+ * @return ArrowSchema generated from `input`
+ */
+std::unique_ptr<ArrowSchema> to_arrow_schema(cudf::table_view const& input,
+                                             std::vector<column_metadata> const& metadata);
```

Comment:
Any reason this can't be a span instead? `vector const&` is kind of an anti-pattern. 

```suggestion
                                             std::span<column_metadata const> metadata);
```

Reply:
Mostly i was just following the existing pattern (look at `to_arrow` which already uses `vector const&`).  I'm not opposed to it being a span, but personally i prefer consistency

Reply:
does libcudf build with C++20? I tried changing this to std::span but I'm getting compiler errors that it can't find `std::span` even with including `<span>`

Reply:
We require C++17 but have a span class for this purpose. Use cudf::host_span like this: https://github.com/rapidsai/cudf/blob/933e32ab9ad8e5057282c48129ddbd745c538967/cpp/include/cudf/concatenate.hpp#L94